pkg/logs/util/windowsevent/message.go

```diff
@@ -48,20 +49,27 @@ func (m *Message) SetContent(content []byte) {
 
 // MapToMessage packages a Map into either an unstructured message.Message or a structured one.
 func MapToMessage(m *Map, source *sources.LogSource, processRawMessage bool) (*message.Message, error) {
+	// Check if the message was truncated by looking for the truncated flag
+	isTruncated := strings.Contains(m.GetMessage(), truncatedFlag)
+
 	// old behaviour using an unstructured message with raw data
 	if processRawMessage {
 		jsonEvent, err := m.Json()
 		if err != nil {
 			return nil, err
 		}
-		return message.NewMessageWithSource(jsonEvent, message.StatusInfo, source, time.Now().UnixNano()), nil
+		msg := message.NewMessageWithSource(jsonEvent, message.StatusInfo, source, time.Now().UnixNano())
+		msg.ParsingExtra.IsTruncated = isTruncated
```

Comment:
The fact that we have to copy this `ParsingExtra` in so many places with this change has me wondering if we should elevate the "truncated" attribute to a top-level attribute on `Message` to make it a "first-class" property of messages.

Granted, this is likely to be a more invasive change. I'm just concerned that the current change is brittle and likely to be accidently broken when someone creates a new message in some place and forgets to copy `ParsingExtra`.